src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs

```diff
@@ -124,6 +125,42 @@ private void AddListed()
                 _metadata["listed"] = listed;
             }
 
+            private void AddSemVerLevelKey()
+            {
+                var version = (string)_catalog["verbatimVersion"];
+                if (version != null)
+                {
+                    NuGetVersion packageOriginalVersion;
+                    if (NuGetVersion.TryParse(version, out packageOriginalVersion))
+                    {
+                        if (packageOriginalVersion.IsSemVer2)
+                        {
+                            _metadata["semVerLevelKey"] = "2";
+                            return;
+                        }
+                    }
+                }
+
+                var dependencyGroups = _reader.GetPackageDependencies().ToList();
+                foreach (var dependencyGroup in dependencyGroups)
+                {
+                    if (dependencyGroup.Packages.Any())
+                    {
+                        // Add packages list
```

Comment:
Not sure what you mean by "adding" packages list. Think the comment can be removed.